src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3687,6 +3700,37 @@ private bool AreEquivalentClosureScopes(SyntaxNode oldScopeOpt, SyntaxNode newSc
 
         #endregion
 
+        #region State Machines
+
+        private void ReportStateMachineRudeEdits(
+            UpdatedMemberInfo updatedInfo, 
+            ISymbol oldMember, 
+            List<RudeEditDiagnostic> diagnostics)
+        {
+            if (!updatedInfo.OldHasStateMachineSuspensionPoint)
+            {
+                return;
+            }
+
+            // only methods and anonymous functions may be async/iterators machines:
+            var stateMachineAttributeQualifiedName = ((IMethodSymbol)oldMember).IsAsync ?
+                "System.Runtime.CompilerServices.AsyncStateMachineAttribute" :
+                "System.Runtime.CompilerServices.IteratorStateMachineAttribute";
+
+            // We assume that the attributes, if exist, are well formed.
+            // If not an error will be reported during emit.
+            if (oldMember.ContainingAssembly.GetTypeByMetadataName(stateMachineAttributeQualifiedName) == null)
```

Comment:
The EnC logic will: https://github.com/dotnet/roslyn/pull/9168/files#diff-e8dfca4da0b5febea0749842089c9424R195